docs/application/native/guides/text-input/voice-control-manager.md

```diff
@@ -0,0 +1,1154 @@
+          /* Error handling */
+  }
+
+  /* Unset */
+  void
+  unset_dialog_request_cb()
+  {
+      int ret;
+      ret = vc_mgr_unset_dialog_request_cb();
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the private data updated callback to get invoked when the voice control engine sets private data to the voice control manager:
+
+  ```c
+  /* Callback */
+  void
+  __private_data_set_cb(const char *key, const char *data, void *user_data)
+  {
+      /* Your code */
+  }
+
+  /* Set */
+  void
+  set_private_data_set_cb()
+  {
+      int ret;
+      ret = vc_mgr_set_private_data_set_cb(__private_data_set_cb, NULL);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Unset */
+  void
+  unset_private_data_set_cb()
+  {
+      int ret;
+      ret = vc_mgr_unset_private_data_set_cb();
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the specific engine result callback to get invoked when the voice control engine sends additional information about the undefined data:
+
+  ```c
+  /* Callback */
+  void
+  __specific_engine_result_cb(const char* engine_app_id, const char* event, const char* result, void *user_data)
+  {
+      /* Your code */
+  }
+
+  /* Set */
+  void
+  set_specific_engine_result_cb()
+  {
+      int ret;
+      ret = vc_mgr_set_specific_engine_result_cb(__specific_engine_result_cb, NULL);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Unset */
+  void
+  unset_specific_engine_result_cb()
+  {
+      int ret;
+      ret = vc_mgr_unset_specific_engine_result_cb();
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the pre recognition result updated callback to get invoked when the voice control engine sets the pre recognition results (partial ASR) to voice control manager:
+
+  ```c
+  /* Callback */
+  void
+  __pre_result_cb(vc_pre_result_event_e event, const char *result, void *user_data)
+  {
+      /* Your code */
+  }
+
+  /* Set */
+  void
+  set_pre_result_cb()
+  {
+      int ret;
+      ret = vc_mgr_set_pre_result_cb(__pre_result_cb, NULL);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Unset */
+  void
+  unset_pre_result_cb()
+  {
+      int ret;
+      ret = vc_mgr_unset_pre_result_cb();
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the all recognition result received callback to get invoked when voice control engine sends the all recognition result to voice control manager.
+  In the callback, the recognized result, recognized text, and the engine message results are included.
+  The recognized result can include more than two voice commands, if the two voice commands have the same command string and are registered by each voice control clients.
+  If you want to select specific command in the recognized result, you can use `vc_mgr_set_selected_results()` in the callback:
```

Comment:
> Is this sentence means that developer can set a callback that includes all the recognition results and that is invoked when all the recognition results received?
Because the name of callback is vc_all_result_cb, I use the all recognition result received callback to match the name of callback.
If your sentence has right meaning, I will fix it.
It is hard to decide whether your sentence is right.
Please share your opinion.

Reply:
@stom-hwang 

- Set all the recognition result received call back to get invoked when the voice control engine sends all recognition results to the voice control manager.  **(This is the same as the original)**

In the callback, the recognized result, recognized text, and the engine message results are included. **(This is the same as the original)**


The recognized result can include more than two voice commands, if those two voice commands have the same command string and are registered by each voice control clients.  **( I have just added those)**

If you want to select specific command in the recognized result, you can use `vc_mgr_set_selected_results()` in the callback: **( This is the same as the original)**


Let me know what made you think like that.